ui/src/app/cohort-search/selection-list/selection-list.component.tsx

```diff
@@ -136,6 +141,55 @@ export class SelectionInfo extends React.Component<SelectionInfoProps, Selection
   }
 }
 
+export class SelectionInfo extends React.Component<SelectionInfoProps, SelectionInfoState> {
+  name: HTMLDivElement;
+  constructor(props: SelectionInfoProps) {
+    super(props);
+    this.state = {truncated: false};
+  }
+
+  componentDidMount(): void {
+    const {offsetWidth, scrollWidth} = this.name;
+    this.setState({truncated: scrollWidth > offsetWidth});
+  }
+
+  get showType() {
+    return ![
+      DomainType.PHYSICALMEASUREMENT.toString(),
+      DomainType.DRUG.toString(),
+      DomainType.SURVEY.toString()
+    ].includes(this.props.selection.domainId);
+  }
+
+  get showOr() {
+    const {index, selection} = this.props;
+    return index > 0 && selection.domainId !== DomainType.PERSON.toString();
+  }
+
+  render() {
+    const {selection, removeSelection} = this.props;
+    const itemName = <React.Fragment>
+      {this.showType && <strong>{typeDisplay(selection)}&nbsp;</strong>}
+      {nameDisplay(selection)} {attributeDisplay(selection)}
+    </React.Fragment>;
+    return <FlexColumn style={styles.selectionItem}>
+      {this.showOr && <div style={{padding: '0.3rem 0rem 0.3rem 1rem'}}>OR&nbsp;</div>}
+      <FlexRow style={{alignItems: 'baseline'}}>
+      <button style={styles.removeSelection} onClick={() => removeSelection()}>
```

Comment:
Thanks. The select icons in the tree are still not enabling after removing a selection, but I can handle that in #3836 since I've made changes to that component anyways.